### ISSUE TITLE
Extend CMake Project to support `make install` and `find_package()`

### ISSUE THREAD

**Issue by thorsten-klein** [hmenke/boost_matheval#13]
In order to build and install matheval the cmake project should use `install()` statements in order to support `make install`, which installs all relevant files. 
Additionally matheval should also provide some cmake EXPORT so that it can be searched by a customer cmake project e.g. via `find_package(boost.matheval)`

**Comment by hmenke** [comment]
Yes, I agree. I should adapt the build system to use [CMakePackageConfigHelpers](https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html). My TODO list is pretty long at the moment, but I'll come back to it.

**Comment by thorsten-klein** [comment]
Hello,
thank you for your fast response 👍🏻 
If I find time I will contribute and create some PR.
